### PR TITLE
Rename rboehm to libgc

### PR DESCRIPTION
**Pull request by jacob-hughes**

### CODE REVIEW

**Comment by ltratt** [comment]
`Multiple candidates for libnatrob.so` -- that old chestnut! I wonder if this is because we're caching something or ... ?

**Comment by jacob-hughes** [comment]
Hmm. Interesting, let me take a look, I think it might have something to do with `cargo update`.

**Comment by jacob-hughes** [comment]
I've pushed a change to buildbot which will do a `cargo clean` before each build. However, if my assumption that `cur_dir/.cargo` lingers between builds is correct, I don't think this will fix the issue.

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
Yeah. I'm sure it's a `.cargo/` problem. There's 2 options: a) I can do a `cargo update` as the first part of the build script, or b) I can login to b13 and do it as a one off here to kick things along. I think _a_ is the correct fix. Thoughts?

**Comment by ltratt** [comment]
I think the right thing to do is to make sure that we don't pick up *any* cache at all, but I think that might be a red herring. In this case, I think the problem is that we have:

```
cargo test
cargo test --release
```

From the buildbot log, `cargo test` completes OK, but we then barf on `cargo test --release`. So I think a `rm -rf target` inbetween `cargo test` and `cargo test --release` might do the trick?

**Comment by jacob-hughes** [comment]
Ok, cargo clean basically does that so I can move that down a line.

**Comment by jacob-hughes** [comment]
bors try

**Comment by ltratt** [comment]
I am not sure that `cargo clean` is fully equivalent to `rm -rf target`.

**Comment by jacob-hughes** [comment]
From the docs: "With no options, cargo clean will delete the entire target directory.". I've observed this locally and that does happen.

I'm a bit confused here as I'm able to follow the build script commands line by line both locally and on bencher13 and it passes without issue. The only lead I have to go on really is that state gets left lying around between builds. I'm not sure how though.

**Comment by jacob-hughes** [comment]
> I'm a bit confused here as I'm able to follow the build script commands line by line both locally and on bencher13 and it passes without issue. The only lead I have to go on really is that state gets left lying around between builds. I'm not sure how though.

By this I mean, cloning into a temp dir and tearing it down again after.

**Comment by jacob-hughes** [comment]
bors try

File: .buildbot.sh [hunk 1]
```diff
@@ -5,6 +5,11 @@ set -e
 export CARGO_HOME="`pwd`/.cargo"
 export RUSTUP_HOME="`pwd`/.rustup"
 
+if [ -f "$CARGO_HOME" ]; then
```
Comment (ltratt) [inline .buildbot.sh]: `-f` is for regular files: I think you need `-d`.

**Comment by jacob-hughes** [comment]
Are you ok if I squash this down (removing the last 3 commits)

**Comment by ltratt** [comment]
You'll want to wait for #30 to complete (assuming it does!), then you can rebase against master and squash.

**Comment by jacob-hughes** [comment]
Squashed

**Pull request merged**